shell.nix

```diff
@@ -0,0 +1,38 @@
+{
+  # This value controls the default version of the bky-as cli that is setup in
+  # the development shell. This value should be updated as the version of
+  # the bky-as cli the example test works against is updated.
+  #
+  # This default value can be overwritten from the command line.
+  bkyas_version ? "e7a2c061da429c66dcfadaf6007e0a4161fea6dc",
+}:
+let
+  nixpkgs = fetchTarball "https://github.com/NixOS/nixpkgs/tarball/nixos-24.11";
+  pkgs = import nixpkgs {
+    config = { };
+    overlays = [ ];
+  };
+
+  mkDevShell = import ./nix/mkDevShell.nix;
+in
+mkDevShell {
+  pkgs = pkgs;
+
+  version = bkyas_version;
+
+  # Note that these package versions are determined by the nixpkgs version
+  # version used to build the shell.
+  devDependencies = [
+    pkgs.git # for project management
+    pkgs.less # for viewing files
+    pkgs.gnumake # for project management
+    pkgs.go # for prepping for building wasm, golangci-lint, and easyjson
+    pkgs.golangci-lint # for linting go files
+    pkgs.easyjson # for generating json marshaling go code
+    pkgs.gotools # for tools like goimports
+    pkgs.jq # for processing data in examples
+    pkgs.nixfmt-rfc-style # for formatting nix files
+    pkgs.tinygo # for building wasm
+    pkgs.toybox # include common unix commands for convenience
```

Comment:
added toybox for common unix commands for convenience in `--pure` mode

Reply:
FWIW, `less` is part of busybox. You can see the full toolset of busy box at https://search.nixos.org/packages?channel=24.11&from=0&size=50&sort=relevance&type=packages&query=busybox

Reply:
I started with busybox but it only works on linux

<img width="214" alt="Screenshot 2025-04-14 at 8 13 51 PM" src="https://github.com/user-attachments/assets/cc57af92-902f-44a9-9994-c5b47b01ad6e" />

Reply:
I think "how do we want to manage the dev relationship with nix-shell" is somewhat of an outstanding question as we are trying to avoid "works on my machine" problems, but also want devs to use it when appropriate. 

I'll go with toybox and less for now, but maybe once we get some more dev interactions with nix we can start a convo in eng agenda